src/librustc_mir/borrow_check/mod.rs

```diff
@@ -2373,24 +2371,56 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         &mut self,
         _: Context,
         (place, span): (&Place<'tcx>, Span),
+        borrow: &BorrowData<'tcx>,
+        scope_tree: &Rc<ScopeTree>,
         end_span: Option<Span>,
     ) {
         let root_place = self.prefixes(place, PrefixSet::All).last().unwrap();
+        let borrow_span = self.mir.source_info(borrow.location).span;
         let proper_span = match *root_place {
             Place::Local(local) => self.mir.local_decls[local].source_info.span,
             _ => span,
         };
-        let mut err = self.tcx
-            .path_does_not_live_long_enough(span, "borrowed value", Origin::Mir);
-        err.span_label(proper_span, "temporary value created here");
-        err.span_label(span, "temporary value dropped here while still borrowed");
-        err.note("consider using a `let` binding to increase its lifetime");
-
-        if let Some(end) = end_span {
-            err.span_label(end, "temporary value needs to live until here");
-        }
 
-        err.emit();
+        match &self.describe_place(place) {
+            Some(description) => {
+                let mut err = self.tcx.path_does_not_live_long_enough(
```

Comment:
This doesn't handle the case where a local needs to live until an `end_span`, aka
```Rust
let x = {
    let y = 0;
    &y
};
```

Reply:
This should now be being handled.